### DISCUSSION TITLE
How to setup `autocommand` for `fastlane` files[Appfile, Fastfile, Pluginfile]?

### DISCUSSION THREAD

**Discussion by aap01** [LunarVim/LunarVim#4235] [Q&A] [answered]
Here is my config

```lua
vim.api.nvim_create_autocmd({ "BufReadPost", "BufNewFile" }, {
  pattern = {
    'Fastfile',
    'Appfile',
    'Matchfile',
    'Pluginfile',
  },
  command = "set filetype=ruby",
  -- callback = function()
  --   vim.cmd("set filetype=ruby")
  -- end,
})
```

It shows the following error if I open     'Fastfile', 'Appfile', 'Matchfile' or 'Pluginfile'

<img width="1380" alt="Screenshot 2023-06-12 at 1 16 36 AM" src="https://github.com/LunarVim/LunarVim/assets/19292809/b68d3639-d188-4c75-a73c-23f136ee3419">

**Comment by gvital3230** [comment]
HI, here is my snippet, that works for me. Another filetype but maybe it would be helpful
```
vim.api.nvim_create_autocmd("BufEnter", {
	pattern = { "*.Jenkinsfile,Jenkinsfile.*,Jenkinsfile" },
	callback = function()
		vim.o.filetype = "groovy"
	end,
})
```

**Reply by aap01** [reply, marked as answer]
Actually, any of the following configurations work fine
```lua
  pattern = {
    'Fastfile', 'Appfile', 'Matchfile', 'Pluginfile',
  },
  command = "set filetype=ruby",
  -- or
  -- callback = function()
  -- vim.cmd("set filetype=ruby")
  -- or
  -- vim.o.filetype = 'ruby'
  -- end,
```
Your reply helped me realize that in my config there was some errorprone config with `null-ls` rather than `autocommand`.
And, I found the problematic config. I was setting up `rubocop` as formatter and it was causing `null-ls` to throw exception for `ruby` files.

```lua
-- formatter
local formatters = require 'lvim.lsp.null-ls.formatters'
formatters.setup {
  { command = 'black', filetypes = { 'python' } },
  -- This line was causing all the mess
  -- { command = 'rubocop', filetypes = { 'ruby' } },
}
```

Thanks for the help!
